stock_lot_import_enhancement/models/stock_lot.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class StockLot(models.Model):
+    _inherit = "stock.lot"
+
+    @api.model
+    def name_search(self, name="", args=None, operator="ilike", limit=100):
+        if args is None:
+            args = []
+        if "|" not in name:
+            return super().name_search(name, args, operator, limit)
+        product_name, lot_name = name.split("|", 1)
```

Comment:
If the product name contains "|", it causes an error.

Reply:
Yes. But I have rarely seen product name contains this kind of character. Please let me know if it can be possible in Japanese product name.